pyccl/ccl_redshifts.i

```diff
@@ -125,14 +122,12 @@ static double call_py_dN_fn(double z, void *py_func_obj, int *status)
 
     // Get Python function object and construct argument list
     func = (PyObject *) py_func_obj;
-    arglist = Py_BuildValue("d", z); // Assumed to take one double arg
-    printf("arglist=%f\n", PyFloat_AsDouble(arglist));
+    arglist = Py_BuildValue("(d)", z); // Assumed to take one double arg
```

Comment:
This was the relevant fix. Notice `"d"` versus `"(d)"`. After a lot of digging I figured out that this does not always return a tuple used to call a function if you build only a single value. The parentheses do the trick.